icinga-app/icinga.cpp

```diff
@@ -374,79 +374,86 @@ int Main(void)
 	} else if (command) {
 		Logger::DisableTimestamp(true);
 #ifndef _WIN32
-		if (command->GetImpersonationLevel() == ImpersonateRoot) {
-			if (getuid() != 0) {
-				Log(LogCritical, "cli", "This command must be run as root.");
-				return 0;
-			}
-		} else if (command && command->GetImpersonationLevel() == ImpersonateIcinga) {
-			String group = Application::GetRunAsGroup();
-			String user = Application::GetRunAsUser();
-	
-			errno = 0;
-			struct group *gr = getgrnam(group.CStr());
-	
-			if (!gr) {
-				if (errno == 0) {
-					Log(LogCritical, "cli")
-					    << "Invalid group specified: " << group;
-					return EXIT_FAILURE;
-				} else {
-					Log(LogCritical, "cli")
-					    << "getgrnam() failed with error code " << errno << ", \"" << Utility::FormatErrorNumber(errno) << "\"";
-					return EXIT_FAILURE;
-				}
-			}
-	
-			if (getgid() != gr->gr_gid) {
-				if (!vm.count("reload-internal") && setgroups(0, NULL) < 0) {
-					Log(LogCritical, "cli")
-					    << "setgroups() failed with error code " << errno << ", \"" << Utility::FormatErrorNumber(errno) << "\"";
-					Log(LogCritical, "cli")
-					    << "Please re-run this command as a privileged user or using the \"" << user << "\" account.";
-					return EXIT_FAILURE;
+
+		switch (command->GetImpersonationLevel()) {
+			case ImpersonateRoot:
+#ifdef ICINGA2_REQUIRE_SUPERUSER
```

Comment:
That was a question by the OP, but no decision has been made on how to implement the issue. ifdef's are not an option here.